apps/web/app/api/v1/client/storage/route.ts

```diff
@@ -42,54 +44,51 @@ export async function POST(req: NextRequest): Promise<NextResponse> {
 
   const { plan } = team;
 
-  const buffer = Buffer.from(fileBuffer);
-
-  const bufferBytes = buffer.byteLength;
-  const bufferKB = bufferBytes / 1024;
-
-  if (plan === "free" && bufferKB > 10240) {
-    return responses.badRequestResponse("Maximum file size for free plan is 10MB, please upgrade your plan");
-  }
-
-  if (plan === "pro" && bufferKB > 1024 * 1024) {
-    return responses.badRequestResponse("Maximum file size for pro plan is 1GB");
-  }
-
-  const uploadPrivateFile = async () => {
-    // if s3 is not configured, we'll upload to a local folder named uploads
-
-    if (!env.S3_ACCESS_KEY || !env.S3_SECRET_KEY || !env.S3_REGION || !env.S3_BUCKET_NAME) {
-      try {
-        await putFileToLocalStorage(fileName, fileBuffer, accessType, environmentId, UPLOADS_DIR);
-
-        return responses.successResponse({
-          uploaded: true,
-          url: new URL(`${WEBAPP_URL}/storage/${environmentId}/${accessType}/${fileName}`).href,
-        });
-      } catch (err) {
-        if (err.name === "FileTooLargeError") {
-          return responses.badRequestResponse(err.message);
-        }
+  return await uploadPrivateFile(fileName, environmentId, fileType, plan);
+}
 
-        return responses.internalServerErrorResponse(err.message);
-      }
-    }
+const uploadPrivateFile = async (
```

Comment:
can we move this to an own file in the `lib` folder of this route to make the file more readable?